tests/e2e/pageobjects/ide/Ide.ts

```diff
@@ -49,6 +49,12 @@ export class Ide {
                 Logger.warn('StaleElementException occured during waiting for IDE. Sleeping for 2 secs and retrying.');
                 this.driverHelper.wait(2000);
                 await this.driverHelper.waitAndSwitchToFrame(By.css(Ide.IDE_IFRAME_CSS), timeout);
+            } else if (err instanceof error.TimeoutError) {
```

Comment:
One improvement:
there is your version
```
    async waitAndSwitchToIdeFrame(timeout: number = TimeoutConstants.TS_SELENIUM_START_WORKSPACE_TIMEOUT) {
        Logger.debug('Ide.waitAndSwitchToIdeFrame');
        try {
            await this.driverHelper.waitAndSwitchToFrame(By.css(Ide.IDE_IFRAME_CSS), timeout);
        } catch (err) {
            if (err instanceof error.StaleElementReferenceError) {
                Logger.warn('StaleElementException occured during waiting for IDE. Sleeping for 2 secs and retrying.');
                this.driverHelper.wait(2000);
                await this.driverHelper.waitAndSwitchToFrame(By.css(Ide.IDE_IFRAME_CSS), timeout);
            } else if (err instanceof error.TimeoutError) {
                Logger.error(`Switching to IDE frame failed after ${timeout} timeout.`)
                throw err;
            } else {
                Logger.error(`Switching to IDE frame failed.`)
                throw err;
            }
        }
    }
 ```
 
 I prefer not to use "else if" in my eyes it is much clear approach
 ```
 async waitAndSwitchToIdeFrame(timeout: number = TimeoutConstants.TS_SELENIUM_START_WORKSPACE_TIMEOUT) {
    Logger.debug('Ide.waitAndSwitchToIdeFrame');
    try {
        await this.driverHelper.waitAndSwitchToFrame(By.css(Ide.IDE_IFRAME_CSS), timeout);
    } catch (err) {
        if (err instanceof error.StaleElementReferenceError) {
            Logger.warn('StaleElementException occured during waiting for IDE. Sleeping for 2 secs and retrying.');
            this.driverHelper.wait(2000);
            await this.driverHelper.waitAndSwitchToFrame(By.css(Ide.IDE_IFRAME_CSS), timeout);
            return;
        } 
        
        if (err instanceof error.TimeoutError) {
            Logger.error(`Switching to IDE frame failed after ${timeout} timeout.`)
            throw err;
        }

        Logger.error(`Switching to IDE frame failed.`)
        throw err;        
    }
}
```
If you not agree wirh rhis approach you can ignore this message.

Reply:
I like this approach more than `if then else`, good catch @Ohrimenko1988

Reply:
Thanks for your idea, I'll not implement it here as it goes just to 7.25.x branch. I'll do it in PR to master.